internal/core/textlayout/linebreak_ascii.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+use super::BreakOpportunity;
+
+pub struct LineBreakIterator<'a> {
+    it: core::str::CharIndices<'a>,
+    last_ch: Option<char>,
+}
+
+impl<'a> LineBreakIterator<'a> {
+    pub fn new(text: &'a str) -> Self {
+        Self { it: text.char_indices(), last_ch: None }
+    }
+}
+
+impl<'a> Iterator for LineBreakIterator<'a> {
+    type Item = (usize, BreakOpportunity);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some((byte_offset, char)) = self.it.next() {
+            let maybe_opportunity = match char {
+                '\u{2028}' | '\u{2029}' => Some(BreakOpportunity::Mandatory), // unicode line- and paragraph separators
```

Comment:
that's not very ascii (contrary to what the module name let us think)

Reply:
That's true heh. Shall I rename the module? `fake_linebreak` maybe?

Reply:
_simple sounds good.

The Unicode algorithm supports a lot more scenarios beyond just breaking after a space. The table increases the size of the binary.